### PR TITLE
Edit profiles-branch

### PR DESCRIPTION
**Pull request by AntoniaGranit**
Finished the patch-endpoints. Please also take a look at route path naming.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -188,9 +186,56 @@ app.get('/:username/:plantId', async (req, res) => {
 });
 
 
+// Edit plant profile
+app.patch('/:username/:plantId/edit', authenticateUser);
+app.patch('/:username/:plantId/edit', async (req, res) => {
```
Comment (SaraBat) [inline server.js:191]: I'm not sure you need a new address for a patch call, my understanding is that you can have 
`app.get('/:username/:plantId/', authenticateUser)`
`app.post('/:username/:plantId', authenticateUser)`
`app.patch('/:username/:plantId', authenticateUser)`

File: server.js [hunk 2]
```diff
@@ -188,9 +186,56 @@ app.get('/:username/:plantId', async (req, res) => {
...
+      if (species) {
+        plant.species = species; // Change species
+      }
+      if (imageUrl) {
+        plant.imageUrl = imageUrl; // Change plant photo
+      }
+      if (birthday) {
+        plant.birthday = birthday; // Change birthday
+      }
+
```
Comment (SaraBat) [inline server.js:213]: does this work if you're only changing one thing? `platname` for example?

File: server.js [hunk 3]
```diff
@@ -188,9 +186,56 @@ app.get('/:username/:plantId', async (req, res) => {
...
+  }
+});
+
+
+
 // Delete plant
-app.delete('/plant/:plantId', authenticateUser);
-app.delete('/plant/:plantId', async (req, res) => {
+app.delete('/:username/:plantId/delete', authenticateUser);
+app.delete('/:username/:plantId/delete', async (req, res) => {
```
Comment (SaraBat) [inline server.js:238]: could it just be
app.delete('/:username/:plantId/', authenticateUser) ?

Comment (SaraBat) [reply server.js:238]: I see now you tried beforehand

File: server.js [hunk 4]
```diff
@@ -221,15 +266,30 @@ app.delete('/plant/:plantId', async (req, res) => {
 
 
 // Get user's garden
-app.get("/:username/garden", authenticateUser);
-app.get("/:username/garden", async (req, res) => {
+app.get("/users/:username/garden", authenticateUser);
+app.get("/users/:username/garden", async (req, res) => {
+  try {
```
Comment (SaraBat) [inline server.js:271]: we need to match this in the frontend to avoid problems

File: server.js [hunk 5]
```diff
@@ -322,6 +382,51 @@ app.get('/:username', async (req, res) => {
...
 
+// Edit user profile
+app.patch("/:username/edit", authenticateUser);
+app.patch("/:username/edit", async (req, res) => {
+  try {
+    const accessToken = req.header("Authorization");
+    const user = await User.findOne({ accessToken: accessToken });
+    const username = req.params.username;
+    if (user.username === username) {
+      const { username, email, password } = req.body;
```
Comment (SaraBat) [inline server.js:393]: we have email here as well

**Review by SaraBat — APPROVED** [review]
We have a few discrepancies between Frontend and Backed repos, mainly paths and deciding if we're asking for an email or not

File: server.js (hunk 1 above)
Comment (AntoniaGranit) [reply server.js:191]: Good point! I'll update this in master

File: server.js (hunk 2 above)
Comment (AntoniaGranit) [reply server.js:213]: Yes, as long as it's a .patch-endpoint and not .put you can change only one thing.

File: server.js (hunk 5 above)
Comment (AntoniaGranit) [reply server.js:393]: This is so you can edit your email address, but we can also remove the entire email property.

**Pull request merged**